topicexplorer/prep.py

```diff
@@ -381,78 +371,839 @@ def get_low_filter(c, words=None, items=None, counts=None):
-            if last_row < (filtered_counts < bin).sum() <= len(filtered_counts):
-                percentage = (old_div(counts[counts <= bin].sum(), float(c.original_length)))
-                print("{0:>5.0f}x".format(bin).rjust(8), end=' ')
-                print('{0:2.1f}%'.format(percentage * 100).rjust(8), end=' ')
-                print((u'\u2588' * int(percentage * 36)).ljust(36), end=' ')
-                print("  {0:0.0f} words".format((filtered_counts <= bin).sum()).rjust(14), end=' ')
-                print("<= {0:>5.0f}x".format(bin).ljust(8))
-                if (filtered_counts < bin).sum() == len(filtered_counts):
-                    break
-            last_row = (filtered_counts >= bin).sum()
-
-
-        print(' ' * 17, "{} total occurrences".format(counts.sum()).ljust(36), end=' ')
-        print('{} words total'.format(get_mask(c, words).sum()).rjust(20))
-        print('')
+    chart = ""
+
+    bin_counts, bins = np.histogram(counts[counts.argsort()[::-1]], bins=bins)
+    chart += "{0:>8s} {1:>8s} {2:<36s} {3:>14s} {4:>8s}".format("Rate", 'Bottom', '% of corpus', "# words", "Rate") + "\n"
+    last_row = 0
+    for bin, count in zip(bins, np.cumsum(bin_counts)):
+        filtered_counts = counts[get_mask(c, words)]
+        if last_row < (filtered_counts < bin).sum() <= len(filtered_counts):
+            percentage = (old_div(counts[counts <= bin].sum(), float(c.original_length)))
+            chart += "{0:>5.0f}x".format(bin).rjust(8)
+            chart += '{0:2.1f}%'.format(percentage * 100).rjust(9)
+            chart += " " + (u'\u2588' * int(percentage * 36)).ljust(36)
+            chart += "{0:0.0f} words".format((filtered_counts <= bin).sum()).rjust(15)
+            chart += " <={0:>5.0f}x".format(bin).ljust(8) + "\n"
+            if (filtered_counts < bin).sum() == len(filtered_counts):
+                break
+        last_row = (filtered_counts >= bin).sum()
+
+    chart += (' ' * 18) + "{} total occurrences".format(counts.sum()).ljust(34)
+    chart += '{} words total'.format(get_mask(c, words).sum()).rjust(20)
+    return chart
+
+def get_low_filter_stops(c, words=None, items=None, counts=None, num=None):
+    import numpy as np
+    input_filter = num
+    valid = True
+    try:
+        candidates = get_candidate_words(c, -input_filter, words=words, items=items, counts=counts)
+        places = np.in1d(c.words, candidates)
+        places = dict(zip(candidates, np.where(places)[0]))
+        candidates = sorted(candidates, key=lambda x: counts[places[x]])
+        filtered_counts = counts[get_mask(c, words)]
+
+        filtered = ""
+        filtered += "Filter will remove " + str(filtered_counts[filtered_counts <= input_filter].sum()) + " tokens"
+        filtered += "of these " + str(len(filtered_counts[filtered_counts <= input_filter])) + " words: "
+        filtered += u' '.join(candidates)
 
+
+        if len(candidates) == len(c.words):
+            valid = False
```

Comment:
again, introduce a ValueError that can be `try-except`-ed rather than a new variable to test.